Add Musca-S1 SoC and board support (#27607)
zephyrproject-rtos/zephyr: boards/arm/v2m_musca_s1/doc/index.rst

Flashing
========

DAPLink
---------

V2M Musca-S1 provides:

- A USB connection to the host computer, which exposes a Mass Storage and an
  USB Serial Port.
- A Serial Flash device, which implements the USB flash disk file storage.
- A physical UART connection which is relayed over interface USB Serial port.

This interfaces are exposed via DAPLink which provides:

- Serial Wire Debug (SWD).
- USB Mass Storage Device (USBMSD).
- UART.
- Remote reset.

For more details please refer
to the `DAPLink Website`_.

Building a secure only application
----------------------------------

You can build applications in the usual way. Here is an example for
the :ref:`hello_world` application.

.. zephyr-app-commands::
   :zephyr-app: samples/hello_world
   :board: v2m_musca_s1
   :goals: build

Open a serial terminal (minicom, putty, etc.) with the following settings:

- Speed: 115200
- Data: 8 bits
- Parity: None
- Stop bits: 1

Reset the board, and you should see the following message on the corresponding
serial port:

.. code-block:: console

   Hello World! musca_s1

Building a secure/non-secure with Trusted Firmware
--------------------------------------------------

> The process requires five steps:

Reviewer: Some mention of the `v2m_musca_s1_nonsecure` board option should probably be added here as the correct board name to use with TF-M.

Author: Done.

1. Build Trusted Firmware-M (TF-M).
2. Import it as a library to the Zephyr source folder.
3. Build Zephyr with a non-secure configuration.
4. Merge the two binaries together and sign them.
5. Concatenate the bootloader with the signed image blob.

In order to build tfm please refer to `Trusted Firmware-M Guide`_.
Follow the build steps for AN521 target while replacing the platform with

> ``-DTARGET_PLATFORM=MUSCA_S1`` and compiler (if required) with ``-DCOMPILER=GNUARM``

Reviewer: This seems to be out of date, and should be updated for the recent build system changes around TF-M 1.2.

Author: Done.

- AHB5 TrustZone Peripheral Protection Controller (PPC).
- Implementation-Defined Attribution Unit (IDAU).

For more details please refer to `Corelink SSE-200 Subsystem`_.

Flashing
========

DAPLink
---------

V2M Musca-S1 provides:

- A USB connection to the host computer, which exposes a Mass Storage and an
  USB Serial Port.
- A Serial Flash device, which implements the USB flash disk file storage.
- A physical UART connection which is relayed over interface USB Serial port.

This interfaces are exposed via DAPLink which provides:

- Serial Wire Debug (SWD).
- USB Mass Storage Device (USBMSD).
- UART.
- Remote reset.

For more details please refer
to the `DAPLink Website`_.

Building a secure only application
----------------------------------

You can build applications in the usual way. Here is an example for
the :ref:`hello_world` application.

.. zephyr-app-commands::
   :zephyr-app: samples/hello_world
   :board: v2m_musca_s1
   :goals: build

Open a serial terminal (minicom, putty, etc.) with the following settings:

- Speed: 115200
- Data: 8 bits
- Parity: None
- Stop bits: 1

Uploading an application to V2M Musca-S1
----------------------------------------

> To upload the :ref:`hello_world` application to the board no extra steps are
> required, you can upload directly the ``build/zephyr/zephyr.hex`` build
> artefact. However, in general the applications must be converted to Intel's hex
> format before being flashed to a V2M Musca-S1. An optional bootloader can also
> be prepended to the image.
> The eMRAM base address alias is 0xA000000, and the QSPI flash base address alias
> is 0x0. The image offset is calculated by adding the flash offset to the
> bootloader partition size (when there is one).

Reviewer: ```suggestion
To upload the :ref:`hello_world` application to the board, no extra steps are
required. You can directly upload ``build/zephyr/zephyr.hex``, which is
generated by Zephyr's build system.

In other situations, applications must first be converted to Intel's hex
format before being flashed to a V2M Musca-S1. An optional bootloader can also
be prepended to the image.

The eMRAM base address alias is ``0xA000000``, and the QSPI flash base address
alias is ``0x0``. The image offset is calculated by adding the flash offset to the
bootloader partition size (when there is one).
```

> A third-party tool (srecord) can be used to concatenate the images and generate
> the Intel formatted hex image.
> For more information refer to the `Srecord Manual`_.

Reviewer: ```suggestion
A third-party tool (srecord) can be used to concatenate the images and generate
the Intel formatted hex image.

For more information refer to the `Srecord Manual`_.
```

----------------------------------------------------------

The process requires five steps:

1. Build Trusted Firmware-M (TF-M).
2. Import it as a library to the Zephyr source folder.
3. Build Zephyr with a non-secure configuration.
4. Merge the two binaries together and sign them.
5. Concatenate the bootloader with the signed image blob.

In order to build tfm please refer to `Trusted Firmware-M Guide`_.
Follow the build steps for AN521 target while replacing the platform with
``-DTFM_PLATFORM=musca_s1`` and compiler (if required) with
``-DTFM_TOOLCHAIN_FILE=toolchain_GNUARM.cmake``.

Copy over TF-M as a library to the Zephyr project source and create a shortcut
for the secure veneers and necessary header files. All files are in the install
folder after TF-M built.

Building the TF-M integration sample for Musca-S1
-------------------------------------------------

The TF-M integration samples can be run using the ``v2m_musca_s1_nonsecure``
target. Please make sure all the requirements listed in the sample's
description are met before building.

.. zephyr-app-commands::
   :zephyr-app: samples/tfm_integration/psa_level_1
   :board: v2m_musca_s1_nonsecure
   :goals: build

> To upload the build artefact to the board first connect the Musca-S1 to your
> computer using the USB port, power it on and copy the ``build/tfm_zephyr.hex``
> file onto the newly appeared MUSCA_S mass storage device (for a more detailed
> description of these steps, please read the 'Uploading an application
> to V2M Musca-S1' section). Once it is done, you may reset the board.
> The ``tfm_zephyr.hex`` file was generated by concatenating the signed TF-M and
> Zephyr binaries with the MCUboot image and converting it to Intel's hex format.
> These steps are all performed automatically by CMake.

Reviewer: ```suggestion
To upload the build artifact to the board, first connect the Musca-S1 to your
computer using the USB port, power it on and copy the ``build/tfm_zephyr.hex``
file onto the MUSCA_S mass storage device. (For a more detailed
description of these steps, please read the 'Uploading an application
to V2M Musca-S1' section.)

Once the file transfer has completed, you may reset the board.

The ``tfm_zephyr.hex`` file was generated by concatenating the signed TF-M and
Zephyr binaries with the MCUboot image, and converting it to Intel's hex format.
These steps are all performed automatically by CMake.
```
